test/libsolidity/syntaxTests/functionTypes/comparison_of_function_types_eq.sol

```diff
@@ -7,5 +7,7 @@ contract C {
     }
 }
 // ----
+// Warning 3075: (73-79): Comparison of internal function pointers can yield unexpected results in the legacy pipeline with the optimizer enabled, and will be disallowed entirely in the next breaking release.
+// Warning 3075: (147-153): Comparison of internal function pointers can yield unexpected results in the legacy pipeline with the optimizer enabled, and will be disallowed entirely in the next breaking release.
```

Comment:
The `comparison_of_function_types_` seem to randomly use external or internal functions. Since these work differently and have different checks, we should really split them into separate cases for external and internal.

Reply:
Are you sure? How can you tell?

Reply:
Because I looked into the files :) I mean, not just this file. That's a group of tests with this prefix.

Reply:
Yeah, but like where does the randomness come in - the functions in this case are internal, no? That's what I'm confused about? There's the `*_lt_1.sol` test, which has
```
contract C {
    function f() public returns (bool ret) {
        return this.f < this.f;
    }
}
```
but these are external function pointer comparisons, so unrelated to the warning.

Reply:
There's no randomness :) I meant "random" in the casual sense of the word - these tests compare either external or internal pointers without any specific pattern:
- `comparison_of_function_types_eq.sol`: compares internal functions
- `comparison_of_function_types_gt_1.sol`: compares external functions
- `comparison_of_function_types_gt_2.sol`: compares internal functions
- `comparison_of_function_types_internal_eq_1.sol`: compares internal functions (and looks like a duplicate of `comparison_of_function_types_eq.sol`)
- `comparison_of_function_types_internal_eq_2.sol`: compares internal pointers
- `comparison_of_function_types_lt_1.sol`: compares external functions
- `comparison_of_function_types_lt_2.sol`: compares internal functions

I mean, looking closer now I see that do we have both internal and external at least for `<` and `>` (still, we should rename them), but `==` is only being tested for internal.

Reply:
In that case, I'd be fine with adding a test case or two for external function pointer comparisons, but I'd delay the renaming of the existing tests for after the release (another PR), so as not to bloat this one. Is that OK?

Reply:
> but these are external function pointer comparisons, so unrelated to the warning.

The relation to the warning is that internal function comparisons are going away, but that does not mean that all tests using such comparisons no longer make sense. Some of them should be switched to use external comparisons.

In this particular case I think that the assumption was basically that the external and internal comparisons work the same way and covering both was not necessary. Now that we introduce a change in behavior we need to cover both cases.

Reply:
> so as not to bloat this one. Is that OK?

Depends. If you want to do some more renaming, including unrelated tests, a separate PR makes sense. But if you're only planning to handle these, I'd still to it here. You can always put it in a separate commit and it's still related to the PR.

Reply:
All of the comparison test (syntaxTests) *are* appropriately named now, are they not? At least the ones that I've created/altered.

Reply:
Well, there's still no external test for `==`.

But ok, if this change does not apply to `<` and `>` after all, they're not related. I'd still rename them though.

Reply:
Is [this](https://github.com/ethereum/solidity/pull/14784/files#diff-d0429c91263151c9e5a30e5486c6767b628a8678558163c5f7fa65488c119a26) not a comparison of external function pointers?

Reply:
Ah, right. I missed it somehow. I guess I expected them next to each other, which would be the case if  `_internal` and `_external` were the end :)